bcipy/signal/model/mach_learning/demo/demo_density_estimation.py

```diff
@@ -21,35 +20,30 @@ def _demo_kde():
     x_plot = np.linspace(-5, 10, 1000)[:, np.newaxis]
 
     # generate a dummy density function to sample data from
-    true_dens = (0.3 * norm(0, 1).pdf(x_plot[:, 0])
-                 + 0.7 * norm(5, 1).pdf(x_plot[:, 0]))
+    true_dens = 0.3 * norm(0, 1).pdf(x_plot[:, 0]) + 0.7 * norm(5, 1).pdf(x_plot[:, 0])
 
     fig, ax = plt.subplots()
-    ax.fill(x_plot[:, 0], true_dens, fc='black', alpha=0.2,
-            label='input distribution')
+    ax.fill(x_plot[:, 0], true_dens, fc="black", alpha=0.2, label="input distribution")
 
     # thumb up rule for bandwidth selection
-    bandwidth = 1.06 * min(
-        np.std(x), iqr(x) / 1.34) * np.power(x.shape[0], -0.2)
+    bandwidth = 1.06 * min(np.std(x), iqr(x) / 1.34) * np.power(x.shape[0], -0.2)
 
     # try different kernels and show how the look like
-    for kernel in ['gaussian', 'tophat', 'epanechnikov']:
-        kde = KernelDensityEstimate(kernel=kernel, bandwidth=bandwidth,
-                                    num_cls=1)
+    for kernel in ["gaussian", "tophat", "epanechnikov"]:
+        kde = KernelDensityEstimate(kernel=kernel, bandwidth=bandwidth, num_cls=1)
         kde.fit(x, y)
         log_dens = kde.list_den_est[0].score_samples(x_plot)
-        ax.plot(x_plot[:, 0], np.exp(log_dens), '-',
-                label="kernel = '{0}'".format(kernel))
+        ax.plot(x_plot[:, 0], np.exp(log_dens), "-", label="kernel = '{0}'".format(kernel))
 
     ax.text(6, 0.38, "N={0} points".format(n))
 
-    ax.legend(loc='upper left')
-    ax.plot(x[:, 0], -0.005 - 0.01 * np.random.random(x.shape[0]), '+k')
+    ax.legend(loc="upper left")
+    ax.plot(x[:, 0], -0.005 - 0.01 * np.random.random(x.shape[0]), "+k")
 
     ax.set_xlim(-4, 9)
     ax.set_ylim(-0.02, 0.4)
     plt.show()
-    print('KDE Flows!')
+    print("KDE Flows!")
```

Comment:
There is no rule for single or double quotes re: PEP, but the rest of the code leans toward single unless punctation is involved.

Reply:
I was noticing this as well. We haven't discussed yet as a team or given any guidelines here. The Python literature doesn't seem to give any recommendations beyond being consistent per project. The Ruby community prefers single quotes for the most part, and reserves the use of double quotes for strings with interpolation. This tends to work well and I believe is my preference.

Reply:
This change comes from black - my preference is for whatever is consistent and automatable. Since I was using black in another project, I had that enabled in VS Code, but I'm happy to switch to any other autoformatter so that we can all share a spec. 

The reason I really liked black is that it operates on the AST to guarantee that it maintains program correctness, it runs super fast, and it takes a strong stance to pre-empt all discussion related to formatting (such as single vs double quote, argument indentation, docstring formatting, etc etc), such that the defaults will always adhere to PEP guidelines while preventing the need to choose in ambiguous situations. As pointed out, there's no no PEP recommendation regarding `'` vs `"` so black just picks one and sticks to it.

I guess what I'm saying is, I'd be happy to pay bribes in snacks or beer to try using black (for which the only non-default setting I use is `--line-length=120`), but I would also be just as happy to use yapf or autopep8 or whatever works, provided we can share a config file that runs in VS Code.

I'd especially like to return to this topic in the context of CI/CD. Really in this case, my failure to adhere to a pre-existing standard (whether it was a stated rule or not) could easily be caught as an error on my local machine or on the build server, thus saving time for everyone.

Sorry for the long-winded reply, and especially sorry if I have missed some linting/formatting guidelines somewhere in the project!

Reply:
I've used both black and yapf; I don't really have a preference so we should definitely discuss during retro and make a decision that's stated clearly somewhere. I know these formatting discussions can provide for a frustrating experience. We currently have a Makefile with a `lint` task that runs `autopep8`, but as you mentioned there are many ways of adhering to pep8 and this is insufficient by itself.

Also, we have a Pivotal Tracker ticket for CI/CD  (https://www.pivotaltracker.com/story/show/175336031) and it's something we would like to get into place, so maybe we should prioritize this higher. Feel free to add comments and suggestions to the ticket.

Reply:
Sounds good. For now I just tried to revert these formatting changes in the demos, and also incorporate the suggested f-string changes/removing a few odds-and-ends that were left in the demos. 

I added a comment on the CI/CD ticket and would be happy to take a crack at that sometime. (I actually would have tried it out late one night, but realized that I can't experiment with Travis without certain permissions on the repo. I might start by forking the repo and playing around, and then I could report my findings)